app/views/components/docs/modal_dialogue.yml

```diff
@@ -0,0 +1,46 @@
+name: Modal dialogue
+description: Shows only one section, with no other navigation options, until the user finishes the immediate task
+body: |
+  Use where the application has gotten into a state from which it shouldn’t or can’t proceed without input from the user.
```

Comment:
I don't have much context or research about when modals should be used, so that's the reason I only scoped it for our project's needs until this we find that this is working and it's potentially valuable in other applications. Otherwise happy to add other cases.